chapter_12_pracic/task5.cpp

```diff
@@ -0,0 +1,114 @@
+// Create a time class that includes integer member values for hours, minutes, and seconds.
+// Make a member function get_time() that gets a time value from the user, and a function
+// put_time() that displays a time in 12 :59 : 59 format.Add error checking to the get_time()
+// function to minimize user mistakes.This function should request hours, minutes, and seconds separately, and check each one for ios error status flagsand the correct range.Hours
+// should be between 0 and 23, and minutesand seconds between 0 and 59. Don�t input these
+// values as strings and then convert them; read them directly as integers.This implies that
+// you won�t be able to screen out entries with superfluous decimal points, as does the
+// ENGL_IO program in this chapter, but we�ll assume that�s not important.
+// In main(), use a loop to repeatedly get a time value from the user with get_time() and
+// then display it with put_time(), like this:
+// Enter hours : 11
+// Enter minutes : 59
+// Enter seconds : 59
+// time = 11 : 59 : 59
+// Do another(y / n) ? y
+// Enter hours : 25
+// Hours must be between 0 and 23
+// Enter hours : 1
+// Enter minutes : 10
+// Enter seconds : five
+// Incorrect seconds input
+// Enter seconds : 5
+// time = 1 : 10 : 05
+
+#include <iostream>
+#include <fstream>
+
+class user_time
+{
+private:
+	int hours;
+	int minutes;
+	int seconds;
+public:
+	void get_time();
+	void show_time() const;
+};
+
+int main()
+{
+	char ch = 'y';
+
+	user_time time;
+	do
+	{
+		time.get_time();
+		time.show_time();
+		std::cout << "again? y/n";
+		std::cin >> ch;
+	} while (ch == 'y');
+
+	return EXIT_SUCCESS;
+}
+
+void user_time::get_time()
+{
+	int limit_hours = 23;
```

Comment:
it could be marked as const

Reply:
Fixed in https://github.com/Dmitriy-Romanchuk/master/pull/12/commits/fbd708b5a089fe13ce20ba7ba105e50cc1ee4e9c